x-pack/plugins/observability/public/hooks/slo/use_clone_slo.ts

```diff
@@ -33,34 +32,35 @@ export function useCloneSlo() {
     },
     {
       onMutate: async ({ slo, originalSloId }) => {
-        // Cancel any outgoing refetches (so they don't overwrite our optimistic update)
-        await queryClient.cancelQueries(sloKeys.lists());
+        await queryClient.cancelQueries({ queryKey: sloKeys.lists(), exact: false });
 
-        const latestQueriesData = (
-          queryClient.getQueriesData<FindSLOResponse>(sloKeys.lists()) ?? []
-        ).at(0);
-        const [queryKey, data] = latestQueriesData ?? [];
+        const queriesData = queryClient.getQueriesData<FindSLOResponse>({
+          queryKey: sloKeys.lists(),
+          exact: false,
+        });
+        const [queryKey, previousData] = queriesData?.at(0) ?? [];
 
-        const originalSlo = data?.results?.find((el) => el.id === originalSloId);
+        const originalSlo = previousData?.results?.find((el) => el.id === originalSloId);
         const optimisticUpdate = {
-          ...data,
+          page: previousData?.page ?? 1,
+          perPage: previousData?.perPage ?? 25,
+          total: previousData?.total ? previousData.total + 1 : 1,
           results: [
-            ...(data?.results ?? []),
+            ...(previousData?.results ?? []),
             { ...originalSlo, name: slo.name, id: uuidv1(), summary: undefined },
           ],
-          total: data?.total ? data.total + 1 : 1,
         };
 
-        // Optimistically update to the new value
-        queryClient.setQueryData(queryKey ?? sloKeys.lists(), optimisticUpdate);
+        if (queryKey) {
+          queryClient.setQueryData(queryKey, optimisticUpdate);
+        }
```

Comment:
I wrongly thought that falling back on a wider query key (lists() instead of list()) would update all the children queries data 😅